### DISCUSSION TITLE
How to change the mode (Video, Photo, Timelapse)

### DISCUSSION THREAD

**Discussion by takerfume** [gopro/OpenGoPro#86] [Q&A] [answered]
Hi, I am happy to use OpenGoPro in my project.

Would you please tell me how to change the mode (Video, Photo, Timelapse) and how to check the mode by Python with BLE API?

I set shutter on according to the [python BLE tutorial no.2](https://gopro.github.io/OpenGoPro/tutorials/python/send-ble-commands).
When I set shutter on while mode is video, video was captured.
When I set shutter on while mode is photo, photo was captured.

I would like to write code to check the mode, change the mode if necessary and capture video for sure.

**Comment by tcamise-gpsw** [comment, marked as answer]
Hello. The easiest way to do this is to configure the Preset that you want to use. You can do this via [BLE](https://gopro.github.io/OpenGoPro/tutorials/python/send-ble-commands#load-preset) or [WiFi](https://gopro.github.io/OpenGoPro/tutorials/python/send-wifi-commands#load-preset).

You can then use the WifI Get Preset Status [command](https://gopro.github.io/OpenGoPro/tutorials/python/send-wifi-commands#get-preset-status).

Alternatively, you can use BLE to poll (or register for notifications for) the Flatmode ID [status](https://gopro.github.io/OpenGoPro/ble_2_0#status-codes) where the flatmodes are:

```
        VIDEO = 12
        LOOPING = 15
        SINGLE_PHOTO = 16
        NIGHT_PHOTO = 18
        BURST_PHOTO = 19
        TIME_LAPSE_VIDEO = 13
        TIME_LAPSE_PHOTO = 20
        NIGHT_LAPSE_PHOTO = 21
        TIME_WARP_VIDEO = 24
        LIVE_BURST = 25
        NIGHT_LAPSE_VIDEO = 26
        SLO_MO = 27
```

**Reply by gerryday** [reply]
Is there a setting endpoint to hit to set the flatmode? like /gopro/camera/setting?setting={flatmode_setting_id}&option={flatmode_id}?
